routes/index.js

```diff
@@ -0,0 +1,56 @@
+const express = require("express");
+const admin = require("firebase-admin");
+const serviceAccount = require("../serviceAccount.json");
+const router = express.Router();
+const quotes = [];
+
+admin.initializeApp({
+    credential: admin.credential.cert(serviceAccount)
+});
+
+const db = admin.firestore();
+
+function genId(quotes) {
+    return quotes.length > 0 ? Math.max(...quotes.map(quote => quote.id)) + 1 : 1;
+}
+
+// PATCH
+router.patch("/quotes/:id", async (req,res) => {
+    try {
+        quotes.length = 0;
+        const list = await db.collection("quotes").get();
+        list.forEach(doc => quotes.push(doc.data()));
+
+        if (!req.body.quote && !req.body.author){
+            return res.status(400).json({message: "You have to pass a quote and/or an author :/"});
+        } 
+        const q = await db.collection("quotes").doc(req.params.id).get();
+        console.log(q);
+        if (!q.data()){
+            return res.status(404).json({message: "Quote not found :("});
+        }
+        
```

Comment:
visto che l'hai fatto con gli if, fai un if per il caso in cui inserisco entrambi i campi. come l'hai fatto adesso mi modifica o il quote o l'author. manca la modifica di entrambi insieme